src/Client/index.ts

```diff
@@ -20,6 +20,7 @@ import { parseConnectionString } from "./parseConnectionString";
 
 export interface DNSClusterOptions {
   domain: string;
+  port: number;
   nodePreference?: NodePreference;
 }
```

Comment:
This is more of a critisism of  #84, but its not very obvious what this does (aside from the type name)

Prehaps we should change it to:

``` ts
export interface DNSClusterOptions {
  discovery: Endpoint | string;
  nodePreference?: NodePreference;
}
```

But this also highlights that I forgot to add:

``` ts
EventStoreDBClient.singleNode(endpoint, options?, defaultCreds?)
// etc
```

in #84

Reply:
DNS-wise, the `string` value doesn't make sense. It's always an endpoint.

Reply:
(from discussion)
``` ts
export interface DNSClusterOptions {
  discovery: Endpoint;
  nodePreference?: NodePreference;
}
```